kafkatest/broker.go

```diff
@@ -183,7 +184,7 @@ func (p *Producer) Produce(topic string, partition int32, messages ...*proto.Mes
 	if p.ResponseError != nil {
 		return 0, p.ResponseError
 	}
-	off := p.ResponseOffset
+	off := atomic.AddInt64(&p.ResponseOffset, int64(len(messages)))
```

Comment:
Oh, actually, an atomic add isn't good enough here.  Two goroutines could update the offset, but then add the produced messages to the channel in a different order, resulting in mixed up offsets going down the channel.  I think we have to wrap the whole thing in a mutex.